File: flutter/lib/api/models/cep_trigger_payload.dart
/// The translation contract between a CEP plugin and Digia's rendering engine.
///
/// Plugin authors map their CEP's native callback into this struct.
/// Digia core never imports CleverTap, MoEngage, or WebEngage types.
class CEPTriggerPayload {
  // --- CEP Context ---

  /// The CEP's own identifier for this campaign instance.
  /// Opaque to Digia Core — passed through for analytics correlation.
  final String cepCampaignId;

  /// Any additional metadata the CEP passes through (e.g. UTM params,
  /// user segment label, CEP-specific tracking fields).
  /// Core does not interpret these — forwarded as-is in ExperienceEvents.
  final Map<String, dynamic> cepMetadata;

  // --- Digia Fields ---

  /// The coupling key that links this CEP campaign to a Digia campaign.
  /// Core uses this to look up the matching WidgetConfig in CampaignStore.
  final String campaignKey;

  /// Optional runtime variables to interpolate into the WidgetConfig.
  /// e.g. { "user_name": "Priya", "offer_value": "20%" }
  /// Keys must match variable placeholders declared in the Digia dashboard.
  final Map<String, String>? variables;

  // --- Trigger attribution (analytics matrix) ---

  /// How this campaign was triggered: `event` | `screen_view` | `manual` |
  /// `app_open`. Surfaced first-class so first-party analytics can record
  /// `trigger_type` on the experience's first event. Null when the plugin/host
  /// did not classify the trigger.
  final String? triggerType;

  /// For `event`/`screen_view` triggers, the originating event or screen name
  /// (matrix `trigger_event`). Null otherwise.
  final String? triggerEvent;

  const CEPTriggerPayload({
    required this.cepCampaignId,
    required this.cepMetadata,
    required this.campaignKey,
    this.variables,
    this.triggerType,
    this.triggerEvent,
  });
}

File: android/digia-engage/src/main/java/com/digia/engage/CEPTriggerPayload.kt
package com.digia.engage

/**
 * The translation contract between a CEP plugin and Digia's rendering engine.
 *
 * Mirrors [CEPTriggerPayload] in Flutter.
 * Plugin authors map their CEP's native campaign callback into this struct.
 * Digia core never imports CleverTap, MoEngage, or WebEngage types.
 */
data class CEPTriggerPayload(
    /** The CEP's own identifier for this campaign instance. Opaque to Digia core. */
    val cepCampaignId: String,

    /** The coupling key that links this CEP campaign to a Digia campaign config. */
    val campaignKey: String,

    /**
     * Any additional metadata the CEP passes through (e.g. UTM params,
     * user segment label, CEP-specific tracking fields).
     * Core does not interpret these — forwarded as-is in lifecycle events.
     */
    val cepMetadata: Map<String, Any?> = emptyMap(),

    /**
     * Optional runtime variables to interpolate into the campaign config.
     * e.g. mapOf("user_name" to "Priya", "offer_value" to "20%")
     * Keys must match variable placeholders declared in the Digia dashboard.
     */
    val variables: Map<String, String>? = null,
)
